eng/finalize-publish.yml

```diff
@@ -0,0 +1,74 @@
+jobs:
+  - job: Finalize_Publish_Versions
+    displayName: Finalize_Publish_Versions
+    dependsOn:
+    - build_Linux_arm_release
+    - build_Linux_arm64_release
+    - build_Linux_musl_x64_release
+    - build_Linux_rhel6_x64_release
+    - build_Linux_x64_release
+    - build_FreeBSD_x64_release
+    - build_OSX_x64_release
+    - build_Windows_NT_x64_release
+    - build_Windows_NT_x86_release
+    - build_Windows_NT_arm_release
+    - build_Windows_NT_arm64_release
+    pool:
+      name: Hosted VS2017
+    steps:
+    # Initialize tooling
+    - script: build.cmd -- /t:BuildTraversalBuildDependencies
+      displayName: Initialize tooling
+
+    - powershell: |
+        $prefix = "refs/heads/"
+        $branch = "$(Build.SourceBranch)"
+        $branchName = $branch
+        if ($branchName.StartsWith($prefix))
+        {
+          $branchName = $branchName.Substring($prefix.Length)
+        }
+        Write-Host "For Build.SourceBranch $branch, FullBranchName is $branchName"
+        Write-Host "##vso[task.setvariable variable=FullBranchName;]$branchName"
+      displayName: Find true SourceBranchName
+
+    - task: MSBuild@1
+      displayName: Publish (versions)
+      inputs:
+        solution: $(Build.SourcesDirectory)\tests\build.proj
+        platform: x64
+        configuration: Release
+        msbuildVersion: 15.0
+        msbuildArchitecture: x64
+        msbuildArguments: >-
+          /t:UpdatePublishedVersions
```

Comment:
Core-Setup has a step before this that gathers the packages, so this target can inspect them to get the versions.

Reply:
I guess I am missing something.  I thought we would be using the build name/tag with the commit hash to update dotnet/versions.  I'll have to look at what the msbuild task actually does...

Reply:
It makes commits like this, with the all package id/versions from the build: https://github.com/dotnet/versions/commit/57b0602ee7b698e7374674cfe4874d3b374562a4. To do that, it looks at the nuspec within each nupkg the build produced.

Here's the targets file: https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks/PackageFiles/VersionTools.targets

Reply:
Ahh.  When I was consuming this to trigger builds outside MS, I was only looking at the latest file.  Never looked at the whole set...